### ISSUE TITLE
Instance creation should have the same name checking rules as instance rename

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#347]
Originally reported of Google Code with ID 291.

```
Currently, instance rename does check that the passed name is either identical to what we get back from the resolver, or that at least it's a prefix of it.

Instance creation doesn't, and this can result in instance creations for 'foo' to result in an instance called 'bar' being created. Or, in case the instance 'bar' was already on the cluster, this results in very confusing error message (see issue 289).

For consistency, we should add the same checking rules in add, which can be as usual skipped via --no-name-check.
```
Originally added on 2012-10-31 13:20:51 +0000 UTC.

Mentioned Issue with GoogleCode ID 289 has been migrated to GitHub with ID #345.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in commit 233f4bc695f26dc8f6dd5f672dd5723b93d0645b, will be released in 2.6.2.
```
Originally added on 2012-11-01 13:10:14 +0000 UTC.

Added Labels: Type-Enhancement
Changed State: Fixed
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as released the bugs fixed in 2.6.2.
```
Originally added on 2013-01-11 12:13:05 +0000 UTC.

Changed State: Released
